File: dawn/src/main/java/com/lzq/dawn/util/log/LogClass.java
package com.lzq.dawn.util.log;

/**
 * @Name :LogClass
 * @Time :2022/8/2 15:30
 * @Author :  Lzq
 * @Desc :
 */
public final class LogClass {
    public abstract static class IFormatter<T> {
        public abstract String format(T t);
    }

    public interface IFileWriter {
        void write(String file, String content);
    }

    public interface OnConsoleOutputListener {
        void onConsoleOutput(@LogUtils.TYPE int type, String tag, String content);
    }

    public interface OnFileOutputListener {
        void onFileOutput(String filePath, String content);
    }
}

File: dawn/src/main/java/com/lzq/dawn/util/log/LogClass.kt
package com.lzq.dawn.util.log

/**
 * @Name :LogClass
 * @Time :2022/8/2 15:30
 * @Author :  Lzq
 * @Desc :
 */
class LogClass {
    abstract class IFormatter<T> {
        abstract fun format(t: T): String?
    }

    interface IFileWriter {
        fun write(file: String?, content: String?)
    }

    interface OnConsoleOutputListener {
        fun onConsoleOutput(@LogUtils.TYPE type: Int, tag: String?, content: String?)
    }

    interface OnFileOutputListener {
        fun onFileOutput(filePath: String?, content: String?)
    }
}